Sources/Clappr/Classes/Base/Loader.swift

```diff
@@ -1,63 +1,46 @@
 open class Loader {
-    internal(set) open var playbackPlugins: [Plugin.Type] = [AVFoundationPlayback.self, NoOpPlayback.self]
+    
+    static let shared = Loader()
+    
+    internal(set) open var playbacks: [Plugin.Type] = [AVFoundationPlayback.self, NoOpPlayback.self]
     #if os (iOS)
     internal(set) open var containerPlugins: [Plugin.Type] = [PosterPlugin.self, SpinnerPlugin.self]
     #else
     internal(set) open var containerPlugins: [Plugin.Type] = []
     #endif
     internal(set) open var corePlugins = [Plugin.Type]()
-
-    internal(set) open var mediaControl: MediaControl.Type = MediaControl.self
-
-    fileprivate var externalPlugins = [Plugin.Type]()
-
-    public convenience init() {
-        self.init(externalPlugins: [])
-    }
-
-    public init(externalPlugins: [Plugin.Type], options: Options = [:]) {
-        self.externalPlugins = externalPlugins
-
-        loadExternalMediaControl(options)
-
-        if !externalPlugins.isEmpty {
-            addExternalPlugins(externalPlugins)
-        }
-
+    
+    var externalPlugins: [Plugin.Type] = []
+    
+    private init() {
         Logger.logInfo("plugins:" +
-            "\n - playback: \(playbackPlugins.map({ $0.name }))" +
+            "\n - playback: \(playbacks.map({ $0.name }))" +
             "\n - container: \(containerPlugins.map({ $0.name }))" +
-            "\n - core: \(corePlugins.map({ $0.name }))" +
-            "\n - mediaControl: \(mediaControl)", scope: "\(type(of: self))")
-    }
-
-    fileprivate func loadExternalMediaControl(_ options: Options) {
-        if let externalMediaControl = options[kMediaControl] as? MediaControl.Type {
-            mediaControl = externalMediaControl
-        }
+            "\n - core: \(corePlugins.map({ $0.name }))")
     }
-
+    
     open func addExternalPlugins(_ externalPlugins: [Plugin.Type]) {
-        self.externalPlugins = externalPlugins
-        playbackPlugins = getPlugins(.playback, defaultPlugins: playbackPlugins)
+        self.externalPlugins.append(contentsOf: externalPlugins)
+        self.externalPlugins = self.removeDuplicate(self.externalPlugins)
+        playbacks = getPlugins(.playback, defaultPlugins: playbacks)
         containerPlugins = getPlugins(.container, defaultPlugins: containerPlugins)
         corePlugins = getPlugins(.core, defaultPlugins: corePlugins)
     }
-
-    fileprivate func getPlugins(_ type: PluginType, defaultPlugins: [Plugin.Type]) -> [Plugin.Type] {
+    
+    private func getPlugins(_ type: PluginType, defaultPlugins: [Plugin.Type]) -> [Plugin.Type] {
         let filteredPlugins = externalPlugins.filter({ $0.type == type })
         return removeDuplicate(filteredPlugins + defaultPlugins)
     }
-
-    fileprivate func removeDuplicate(_ plugins: [Plugin.Type]) -> [Plugin.Type] {
+    
+    private func removeDuplicate(_ plugins: [Plugin.Type]) -> [Plugin.Type] {
```

Comment:
Instead of removing duplicates after merge we could use a different structure to store the plugins since we just need to merge and consider priorities when actually loading the plugin and merging with the externals.

We could use a dictionary with the plugin `name` as a key. This way when a duplicate key is registered, the latter would replace the existing plugin as expected.